### PR TITLE
JDK-8315921: Invalid CSS declarations in java.lang class documentation

### PR DESCRIPTION
**Pull request by hns**
This change fixes two errors in inline HTML styles in the `java.lang` package: 

 - wrong CSS property name in `java.lang.String` 
 - CSS declaration terminated by colon instead of semicolon in `java.lang.Thread`

Both errors caused the style declarations to be ignored and an error message to be shown in the browser console. 

The bug is `noreg-doc`, I tested the docs in the browser to make sure the error message is gone and the HTML displays as intended.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315921](https://bugs.openjdk.org/browse/JDK-8315921): Invalid CSS declarations in java.lang class documentation (**Bug** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [cf60279a](https://git.openjdk.org/jdk/pull/15762/files/cf60279abd516df4dcba14179f2675f52c23c016)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15762/head:pull/15762` \
`$ git checkout pull/15762`

Update a local copy of the PR: \
`$ git checkout pull/15762` \
`$ git pull https://git.openjdk.org/jdk.git pull/15762/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15762`

View PR using the GUI difftool: \
`$ git pr show -t 15762`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15762.diff">https://git.openjdk.org/jdk/pull/15762.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15762#issuecomment-1721339462)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/Thread.java [hunk 1]
```diff
@@ -189,7 +189,7 @@
  * In the JDK Reference Implementation, the virtual thread scheduler may be configured
  * with the following system properties:
  * <table class="striped">
- * <caption style="display:none:">System properties</caption>
+ * <caption style="display:none;">System properties</caption>
```
Comment (jaikiran) [inline src/java.base/share/classes/java/lang/Thread.java:192]: Just out of curiosity, is that semi-colon even needed, or would `display:none` be enough?

Comment (jaikiran) [reply src/java.base/share/classes/java/lang/Thread.java:192]: Now that I think about it, do you know why the `display:none` is being set here? I have very limited knowledge of CSS, wouldn't the use of `display:none` make the `System Properties` caption text invisible?

**Comment by hns** [comment]
Thanks for the review, @jaikiran. 

I had found the issues in the original commit just by accidental browsing of the API docs and had assumed they were one-off errors. Now searching over OpenJDK sources again for these errors I found a lot more, most of them in `java.text.MessageFormat`. I fixed these in a second commit.

File: src/java.base/share/classes/java/lang/Thread.java (hunk 1 above)
Comment (hns) [reply src/java.base/share/classes/java/lang/Thread.java]: Indeed, the semi-colon is optional at the end of an inline CSS attribute. I removed it in my follow-up commit. 

Regarding the presence of invisible `<caption>` elements in HTML tables: this is indeed questionable. I assume it was based on the notion that a `<caption>` element is required in HTML tables, which is not the case (neither for HTML4 nor HTML5). However, it is quite pervasive (there are around 170 occurrences of `<caption style="display:none">` in OpenJDK source), and one could argue that it serves as an informational comment both in Java and HTML.

**Review by AlanBateman — APPROVED** [review]
I think two of cases where the CSS was terminated with a colon may have been my fault.

**Comment by hns** [comment]
Thanks, I won't blame you, CSS within HTML within JavaDoc comments is weird. :)

**Pull request closed without merging**